thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/validators/DetectionConfigValidator.java

```diff
@@ -104,80 +112,155 @@ public void validateConfig(DetectionConfigDTO detectionConfig) throws IllegalArg
 
     // Validate all the rules
     Set<String> names = new HashSet<>();
     List<Map<String, Object>> ruleYamls = ConfigUtils.getList(detectionYaml.get(PROP_RULES));
-    for (int i = 1; i <= ruleYamls.size(); i++) {
-      Map<String, Object> ruleYaml = ruleYamls.get(i - 1);
+    for (int ruleIndex = 1; ruleIndex <= ruleYamls.size(); ruleIndex++) {
+      Map<String, Object> ruleYaml = ruleYamls.get(ruleIndex - 1);
 
       // Validate detection rules
-      Preconditions.checkArgument(ruleYaml.containsKey(PROP_DETECTION), "In rule no." + (i) + ", detection rule is missing.");
+      Preconditions.checkArgument(ruleYaml.containsKey(PROP_DETECTION),
+          "Detection rule missing for sub-alert " + alertName + " rule no. " + ruleIndex);
       List<Map<String, Object>> detectionRuleYamls = ConfigUtils.getList(ruleYaml.get(PROP_DETECTION));
       for (Map<String, Object> detectionRuleYaml : detectionRuleYamls) {
-        validateRule(detectionRuleYaml, i, "detection", names);
-        names.add(MapUtils.getString(ruleYaml, PROP_NAME));
+        validateRule(alertName, detectionRuleYaml, ruleIndex, "detection", names);
+        names.add(MapUtils.getString(detectionRuleYaml, PROP_NAME));
       }
 
       // Validate filter rules
       if (ruleYaml.containsKey(PROP_FILTER)) {
         List<Map<String, Object>> filterRuleYamls = ConfigUtils.getList(ruleYaml.get(PROP_FILTER));
         for (Map<String, Object> filterRuleYaml : filterRuleYamls) {
-          validateRule(filterRuleYaml, i, "filter", names);
-          names.add(MapUtils.getString(ruleYaml, PROP_NAME));
+          validateRule(alertName, filterRuleYaml, ruleIndex, "filter", names);
+          names.add(MapUtils.getString(filterRuleYaml, PROP_NAME));
         }
       }
     }
 
     // Safety condition: Validate if maxDuration is greater than 15 minutes
     Map<String, Object> mergerProperties = MapUtils.getMap(detectionYaml, PROP_MERGER, new HashMap());
     if (mergerProperties.get(PROP_MAX_DURATION) != null) {
-      Preconditions.checkArgument(MapUtils.getLong(mergerProperties, PROP_MAX_DURATION) >= datasetConfig.bucketTimeGranularity().toMillis(),
+      DatasetConfigDTO datasetConfig = provider
+          .fetchDatasets(Collections.singletonList(metricConfig.getDataset()))
+          .get(metricConfig.getDataset());
+      Preconditions.checkArgument(
+          MapUtils.getLong(mergerProperties, PROP_MAX_DURATION) >= datasetConfig.bucketTimeGranularity().toMillis(),
           "The maxDuration field set is not acceptable. Please check the the document and set it correctly.");
     }
   }
 
+  private void validateCompositeAlertConfig(Map<String, Object> detectionYaml, String parentAlertName) throws IllegalArgumentException {
+    validateBasicAttributes(detectionYaml, parentAlertName);
+    String alertName = (String) detectionYaml.get(PROP_NAME);
+
+    // Validate all compulsory fields
+    Preconditions.checkArgument(detectionYaml.containsKey(PROP_ALERTS),
```

Comment:
Let's set a default prop_alerts if  it is not provided.

Reply:
The default will be set in the template. The validator is agnostic of what metric/dataset/entity the user wants to monitor.